### DISCUSSION TITLE
Keyboard not mapped RedHat 8 v3.x

### DISCUSSION THREAD

**Discussion by jeffrahr** [FreeRDP/FreeRDP#10473] [General]
We have always built freerdp from source for our  RedHat 8 environment with no issues. We have recently attempted to build versions v3.5..1, v3.6.3, and v3.7.0  and have found that the keyboard is not mapping correctly when connecting to remote systems. I have tried passing a kbd string and that does not have any affect. Typing results were when typing in the "search windows" text box
Typing
qwert
Results in
op[] and then the setting window is displayed when "t" is typed
Typing 
7890-=
Resuts in
qwerty

With versions v2.11.2 and v2.11.7 the keyboard maps correctly, so not sure what has changed in v3.x that it is not mapping the keyboard correctly. Too glaring of a problem to be bug, so has to be something in my setup and/or environment.

Any suggestions, will be greatly appreciated.

**Comment by akallabeth** [comment]
sorry, not enough information to help here.
1. you´re using `xfreerdp`, right?
2. your build config might help
3. a build with `-DWITH_DEBUG_KBD=ON` will log the `X11` to `RDP` scancode mapping with `/log-level:debug`  or `/log-filters:com.freerdp.locale:debug`

**Reply by jeffrahr** [reply]
Thank you for the reply.
1. Yes I am using xfreerdp
2. cmake
     -DCMAKE_INSTAL_PREFIX=/opt/local/freerdp/3.6.3
     -GNinja
     -DCHANNEL_URBDRC=ON
     -DWITH_FAAC=OFF
     -DWITH_GSM=ON
     -DWI_CAIRO=ON
     -DWITH_SWCALE=OFF
     -DWITH_FFMPEG=OFF
     -DWITH_DSP_FFMPEG=OFF
3. Added    -DWITH_DEBUG_KBD=ON
   Running with /log-filters:com.freerdp.locale:debug shows that there is no mapping of keys taking place and the keycode that is typed is the same as rdp code
  Type test
   x11 keycode: 1C -> rdp code: [001c]  1C
   x11 keycode: 1C -> rdp code: [001c]  1C
   x11 keycode: 1A -> rdp code: [001a]  1A
   x11 keycode: 1A -> rdp code: [001a]  1A
   x11 keycode: 27 -> rdp code: [0027]  27
   x11 keycode: 27 -> rdp code: [0027]  27
   x11 keycode: 1C -> rdp code: [001c]  1C
   x11 keycode: 1C -> rdp code: [001c]  1C
rdp codes should be 14, 12, 1F, 14

The only other keyboard debug info I am seeing is prior to password prompt:
[kbd_layout_id_from_x_property] : _XKB_RULES_NAMES layout: us, variant:
[find_keyboard_layout_in_xorg_rules]: xkbLayout: us  xkbVariant:

I apologize for any typo's above, this is from an offline system with no exports allowed.

One other thing I didn't mention is that we are building with a module version of gcc (v13.2.0) since it would not build with the onboard gcc.

**Reply by akallabeth** [reply]
@jeffrahr sounds like you are missing some `X11` related development headers in your build (thus I was asking for the `buildconfig`)
* check `libxkbfile-devel` (at least on fedora, no idea for rhel) is installed before building.
* also check the headers were found

**Reply by jeffrahr** [reply]
Thank you for your help. Yes it was missing libxkbfile-devel (same name on rhel),
